models/Performance.js

```diff
@@ -83,97 +70,78 @@ module.exports = class Performance {
 
       const query = client.query(queryString);
 
-      query.on('row', (performance) => {
-        performances.push(this.format(performance, formatString));
+      query.on('row', ({ name, openat, closeat, performdate }) => {
+        performances.push(new Performance(name, openat, closeat, performdate));
       });
+
       query.on('end', () => {
         client.end();
         resolve(performances);
       });
+
       query.on('error', (err) => {
         client.end();
         reject(err);
       });
     });
   }
 
-  findCurrent() {
+  static findNextOrOpenWindow() {
     return new Promise((resolve, reject) => {
       const client = utils.db.createClient();
-      const queryString = 'SELECT * FROM performances WHERE current LIMIT 1';
+      const queryString = 'SELECT * FROM performances WHERE now() < closeAt ORDER BY openAt ASC LIMIT 1';
 
       client.connect();
       client.on('error', (err) => reject(err));
 
       const query = client.query(queryString);
 
-      query.on('row', (result) => {
+      query.on('row', ({ id, name, openat, closeat, performdate }) => {
         client.end();
```

Comment:
Depending on how often you're doing database queries, you may find it advantageous to create a helper function to run database queries. It would take a query string and any parameters and handle connecting, running the query, collecting rows, and closing the client when done, and just return a promise. Rough sketch follows.

```js
// Rough sketch
const query = module.exports = (queryString, params) => new Promise((resolve, reject) => {
    const client = utils.db.createClient();
    client.connect(); // Is connect() synchronous? Something seems a bit fishy. Are you creating a new pool every time or just getting a connected client from the pool?
    client.on('error', (err) => reject(err));
    const response = client.query(queryString, params);
    const rows = [];
    response.on('row', (row) => {
        rows.push(row);
    });
    response.on('end', () => {
        client.end();
        resolve(rows);
    });
    response.on('error', (err) => {
       client.end();
       reject(err);
    });
});

// Usage
query('SELECT now();')
.then((rows) => {
    console.log(rows[0].now);
})
.catch((err) => {
    console.error('Whoops!', err);
});
```